host/xtest/pkcs11_1000.c

```diff
@@ -8016,3 +8016,446 @@ static void xtest_pkcs11_test_1025(ADBG_Case_t *c)
+		{ CKA_DECRYPT, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_EXTRACTABLE, &(CK_BBOOL){ CK_TRUE }, sizeof(CK_BBOOL) },
+		{ CKA_SENSITIVE, &(CK_BBOOL){ CK_FALSE}, sizeof(CK_BBOOL) },
+	};
+
+	CK_ATTRIBUTE *unwrapped_key_template = (t->target_type == CKK_AES) ?
+						unwrapped_aes_key_template :
+						unwrapped_rsa_key_template;
+	CK_ULONG unwrapped_key_template_size = t->target_type == CKK_AES ?
+				ARRAY_SIZE(unwrapped_aes_key_template) :
+				ARRAY_SIZE(unwrapped_rsa_key_template);
+
+	CK_RSA_PKCS_OAEP_PARAMS oaep_params = {
+		CKM_SHA256, CKG_MGF1_SHA256, CKZ_DATA_SPECIFIED, NULL, 0,
+	};
+	CK_RSA_AES_KEY_WRAP_PARAMS rsa_aes_params = {
+		256, &oaep_params,
+	};
+	CK_MECHANISM rsa_aes_key_wrap_mechanism = {
+		CKM_RSA_AES_KEY_WRAP, &rsa_aes_params, sizeof(rsa_aes_params),
+	};
+
+	CK_BYTE aes_val[RSA_AES_MAX_KEY_SIZE] = { 0 };
+	CK_ULONG key_len = 0;
+	CK_ATTRIBUTE aes_get_template[] = {
+		{ CKA_VALUE_LEN, &key_len, sizeof(key_len) },
+		{ CKA_VALUE, aes_val, sizeof(aes_val) },
+	};
+
+	CK_BYTE unwrapped_val[RSA_AES_MAX_KEY_SIZE] = { 0 };
+	CK_ULONG unwrapped_key_len = 0;
+	CK_ATTRIBUTE aes_get_template_unwrapped[] = {
+		{ CKA_VALUE_LEN, &unwrapped_key_len,
+		  sizeof(unwrapped_key_len) },
+		{ CKA_VALUE, unwrapped_val, sizeof(unwrapped_val) },
+	};
+
+	CK_BYTE unwrapped_rsa_modulus[512] = { 0 };
+	CK_BYTE unwrapped_rsa_private_exponent[512] = { 0 };
+	CK_ATTRIBUTE rsa_template_unwrapped[] = {
+		{ CKA_MODULUS, NULL, 0 },
+		{ CKA_MODULUS, unwrapped_rsa_modulus,
+		  sizeof(unwrapped_rsa_modulus) },
+		{ CKA_PRIVATE_EXPONENT, NULL, 0 },
+		{ CKA_PRIVATE_EXPONENT, unwrapped_rsa_private_exponent,
+		  sizeof(unwrapped_rsa_private_exponent) },
+	};
+	CK_BYTE buf[5120] = { 0 };
+	CK_ULONG size = 0;
+
+	assert(t != NULL);
+	assert(t->target_type == CKK_RSA || t->target_type == CKK_AES);
+
+	Do_ADBG_BeginSubCase(c,
+			"Test RSA AES wrap/unwrap of %lu %s key with %lu RSA",
+			target_bits, (t->target_type == CKK_AES) ? "AES" : "RSA",
+			t->key.modulus_len * 8);
+
+	rv = C_CreateObject(session, wrapping_key_template,
+			    ARRAY_SIZE(wrapping_key_template),
+			    &wrapping_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	rv = C_CreateObject(session, unwrapping_key_template,
+			    ARRAY_SIZE(unwrapping_key_template),
+			    &unwrapping_key);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	if (t->target_type == CKK_AES) {
+		rv = C_GenerateKey(session, &cktest_aes_keygen_mechanism,
+				   aes_key_template,
+				   ARRAY_SIZE(aes_key_template),
+				   &target_key);
+	} else {
+		rv = C_CreateObject(session, target_key_template,
+				    ARRAY_SIZE(target_key_template),
+				    &target_key);
+	}
+
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	size = sizeof(buf);
+	rv = C_WrapKey(session, &rsa_aes_key_wrap_mechanism, wrapping_key,
+		       target_key, buf, &size);
+
+	if (!ADBG_EXPECT_CK_OK(c, rv) ||
+	    !ADBG_EXPECT_COMPARE_UNSIGNED(c, size, <=, sizeof(buf)))
+		goto out;
```

Comment:
enforce rv != `CKR_OK` here (so that `test_rsa_aes_wrap()` returns an error code)?
Ditto at line 8308 and 8322.

Reply:
done in [874db63](https://github.com/OP-TEE/optee_test/pull/627/commits/874db6362878c4f73847d78c87fcbd921a9d12b4)